src/odemis/gui/cont/tabs.py

```diff
@@ -2915,9 +2909,12 @@ def Show(self, show=True):
                 f = self._aligner_xy.moveAbs(md[model.MD_FAV_POS_ACTIVE])
                 self._actuator_controller._enable_buttons(False)
                 f.add_done_callback(self._on_align_move_done)
+            if main_data.role == "cryo-secom":
+                self.stage.position.subscribe(self._on_stage_pos, init=True)
         else:
             main_data.is_acquiring.unsubscribe(self._on_acquisition)
             self._aligner_xy.position.unsubscribe(self._on_align_pos)
+            self.stage.position.unsubscribe(self._on_stage_pos)
```

Comment:
No, it shouldn't be in the Show(), as this function is called when the user has just pressed the tab to display it (or hide it). Just put the subscribe in the `__init__()` and the unsubscribe in the `terminate()`.